### ISSUE TITLE
v0.18.0 testing

### ISSUE THREAD

**Issue by Sjors** [bitcoin/bitcoin#15555]
Link to binaries:
* v0.18.0rc4 : https://bitcoincore.org/bin/bitcoin-core-0.18.0/test.rc4/
* ~~v0.18.0rc3 : https://bitcoincore.org/bin/bitcoin-core-0.18.0/test.rc3/~~
* ~~v0.18.0rc2 : https://bitcoincore.org/bin/bitcoin-core-0.18.0/test.rc2/~~
* ~~v0.18.0rc1 : https://bitcoincore.org/bin/bitcoin-core-0.18.0/test.rc1/~~

Let us know which version you tested on which operating system. If you find an issue, please search Github for known issues first and then open a new Github issue.

See [Release Notes](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.18.0-Release-Notes-Draft) for a list of changes, and testing reports for earlier releases ([v0.17.1](https://github.com/bitcoin/bitcoin/issues/14902)), for an idea what to test.

**Comment by Sjors** [comment]
tACK v0.18.0rc1 on macOS 10.14.3:
* tested installing from (signed) DMG
* tested loading wallets
* testing sending coins
* testing receiving coins (notification)

**Comment by kiv06041992** [comment]
![q](https://user-images.githubusercontent.com/5583517/54013238-6efe5f00-4189-11e9-8db1-380531cbdae2.png)

**Comment by promag** [comment]
@kiv06041992 mind sharing relevant configuration?

**Comment by kiv06041992** [comment]
OS: Ubuntu 18.04.2 LTS - x64
CPU: AMD® Fx(tm)-6100 six-core processor × 6 
Memory: 11.6гб
and russian language

0.17.1 is working stable

If you need you may connect towards my computer via TeamViewer

**Comment by makoting** [comment]
bitcoind v0.18.0rc1 is using 100% CPU resources while syncing the blocks. Top, the processes monitoring utility, shows bitcoind-opencon thread is the CPU hog. Ran with -proxy=127.0.0.1:9050.

Debian 9.8 x86_64 Linux GNU as guest VM
VirtualBox 6.0.4, Windows 10 as host
Intel Core i5-8250U CPU

**Comment by hebasto** [comment]
@kiv06041992 
I can try reproduce an issue on Ubuntu with Russian language. Which desktop environment do you use?
Let us know your `bitcoin.conf` and your command-line options if any as well.

On the other hand, you can submit a new issue.

**Comment by jasonzhouu** [comment]
tACK v0.18.0rc1 on macOS 10.14.2:

- tested installing from (signed) DMG
- tested loading wallets
- tested sending coins
- testing receiving coins (notification), testnet receive address:
  - tb1qcn58cs07dxg5292w3zrkvjapzgjdtts5lvx7s2
  - 2NBaJprx17hpMwo1Va15epfAC7iw8zVHuPT

**Comment by kiv06041992** [comment]
@hebasto 
My bitcoin.conf is empty. I use ubuntu is Ubuntu Desktop 18.04.2 (64-bit) https://ubuntu.ru/get

I looked debug.log and saw it.
`2019-03-07T20:04:10Z GUI: Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt.
You must not let any exception whatsoever propagate through Qt code.
If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.

2019-03-07T20:04:10Z 

************************
EXCEPTION: N5boost10filesystem16filesystem_errorE 
filesystem::recursive_directory_iterator directory error: Отказано в доступе 
bitcoin in Runaway exception`

**Comment by MarcoFalke** [comment]
@kiv06041992 Could you please share the lines before that in the debug.log?

**Comment by promag** [comment]
@kiv06041992 looks like your exception is the same as https://stackoverflow.com/a/23135989.

Mind sharing boost version used to build?

**Comment by kiv06041992** [comment]
@MarcoFalke https://raw.githubusercontent.com/kiv06041992/kiv06041992.github.io/master/debug.log

**Comment by MarcoFalke** [comment]
Does this also happen if you run `bitcoind`? If so, could you please run `bitcoind` in `gdb` to get a stack trace of the exception?

**Comment by kiv06041992** [comment]
@promag 
I download 
bitcoin-0.18.0rc1-x86_64-linux-gnu.tar.gz  [https://bitcoincore.org/bin/bitcoin-core-0.18.0/test.rc1/](url)
and put 
bitcoin-0.18.0rc1-x86_64-linux-gnu.tar.gz/bitcoin-0.18.0rc1/bin/bitcoin-qt 
in
/usr/local/bin and run him ./bitcoin-qt
I did not use build.
Only 0.18.0 crashed. v0.17.1 work perfectly

Sorry. I bad speak english but want help

**Comment by promag** [comment]
Please run `bitcoin-qt -nowallet` and see if the error occurs.

**Comment by kiv06041992** [comment]
@promag I ran `bitcoin-qt -nowallet` and waiting
@MarcoFalke I did not run bitcoind. If will `bitcoin-qt -nowallet` crash i will try run `bitcoind` in `gdb`

**Comment by promag** [comment]
I've managed to reproduce the problem (or apparently) in macos:

```sh
# 1. create directory for -walletdir without read access
mkdir /tmp/foo && chmod a-r /tmp/foo

# 2. download and mount https://bitcoincore.org/bin/bitcoin-core-0.18.0/test.rc1/bitcoin-0.18.0rc1-osx.dmg

# 3. run bitcoin-qt and should print a error, but continues running
/Volumes/Bitcoin-Core/Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt -regtest -walletdir=/tmp/foo
/private/tmp/foo: Permission denied

# 4. go to File -> Open Wallet and should segfault
EXCEPTION: N5boost10filesystem16filesystem_errorE
boost::filesystem::directory_iterator::construct: Permission denied: "/private/tmp/foo"
bitcoin in Runaway exception
```

I'm working on a fix.

**Comment by MarcoFalke** [comment]
That fixes the underlying wallet issue, but it seems there is still the gui issue:

```
Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt.
You must not let any exception whatsoever propagate through Qt code.
If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.

**Comment by promag** [comment]
> must not let any exception whatsoever propagate

@MarcoFalke the fix does this?

**Comment by kiv06041992** [comment]
@promag 
> Please run `bitcoin-qt -nowallet` and see if the error occurs.
My node is still working. Thanks.

**Comment by promag** [comment]
@kiv06041992 note that running with `-nowallet` doesn't fix the issue.

**Comment by jonatack** [comment]
Partial tACK bitcoin-0.18.0rc1-x86_64-linux-gnu.tar.gz

PureOS distribution of Linux Debian 4.19.16-1 (2019-01-17) x86_64 GNU/Linux

- tested installing
- tested running bin/bitcoind
- tested running bin/bitcoin-qt, verifying debug windows info/console/network/peers, and running cli commands in the console

**Comment by jonatack** [comment]
Idem for compiling v0.18.0rc1 tagged branch from source.

**Comment by MarcoFalke** [comment]
> You can also report issues you found here

I'd advise against reporting issues here. They should go in their own report. Otherwise it is hard to follow-up on reports when everything is in the same thread.

> See Release Notes for a list of changes

The link points to the file in git, whereas the release notes draft is located in https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.18.0-Release-Notes-Draft

**Comment by Sjors** [comment]
I updated the release note link and changed the wording to encourage new tickets.

**Comment by lucayepa** [comment]
Partially tested rc2 on Debian oldstable (8.11), Debian stable (9.8), Alpine (3.9.2):
- bitcoind compiled from source, tested, used on mainnet with regular p2p activity
- reindexed from scratch
- txindex deleted and rebuilt from scratch

**Comment by farrilis** [comment]
CPU maxing out using 18 rc2 binaries, Debian 9.8, using prune=550 with bitcoin-qt

importmulti -ing several hundred watch only addresses to a non-default wallet apparently triggered this, but subsequently executing with -disablewallet exhibits the same issue (CPU spikes to 100% for 5-10s every 30s)

No such CPU spikes using 0.17.1 (either loading -wallet=mywatchonly or -disablewallet). 0.18.0rc2 does exhibit the described spikes with either -wallet=mywatchonly or -disablewallet. Hence maybe a prune=550 issue.

**Comment by MarcoFalke** [comment]
@farrilis Could you determine which thread is using the CPU? (Maybe with `top` or `htop`)
Alternatively get a `bt` in `gdb` when the spike happens to see the exact code location.

**Comment by farrilis** [comment]
$ gdb ./bitcoin-qt
(gdb) thread apply all bt
(gdb) run

last 3 lines in gdb are:

[New Thread 0x7fff41ffb700 (LWP 2406)]
[Thread 0x7fff51063700 (LWP 2400) exited]
[Thread 0x7fff52866700 (LWP 2395) exited]

not too familiar with gdb, hope that's useful

**Comment by farrilis** [comment]
also, running with gdb seems to permanently peg CPU usage

**Comment by MarcoFalke** [comment]
With `htop` you can select `F5` for tree view and `F2`-"Display options"-[x] thread names to get the thread names.

In `gdb`, you'd have to break before the `bt`. I guess CTRL+C does this.

**Comment by farrilis** [comment]
`htop` is telling me `bitcoin-opencon` is responsible. Peers are neither added or dropped coinciding with the CPU spike

**Comment by MarcoFalke** [comment]
@farrilis Could you do the following:

* Open a terminal window with `htop`
* Open another terminal window with `gdb --args ./bitcoin-qt`
** Type `run` in (gdb)
** Watch the CPU usage of the `opencon` thread in the htop window
** Hit `CTRL+C` in (gdb) when it is at 100%
** Type `thread apply all bt` in (gdb)

**Comment by farrilis** [comment]
Ok:

`Thread 22 (Thread 0x7fff45fb0700 (LWP 2310)):
#0  0x00007ffff637e981 in __GI_ppoll (fds=0x555557c58608, nfds=1, timeout=<optimized out>, sigmask=0x0) at ../sysdeps/unix/sysv/linux/ppoll.c:39
#1  0x00005555562934c1 in ?? ()
#2  0x0000555556294a99 in ?? ()
#3  0x000055555624245f in ?? ()
#4  0x00005555560e2956 in ?? ()
#5  0x00005555560e4c25 in ?? ()
#6  0x00007ffff79bb4a4 in start_thread (arg=0x7fff45fb0700) at pthread_create.c:456
#7  0x00007ffff6387d0f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97

Thread 21 (Thread 0x7fff467b1700 (LWP 2309)):
#0  0x00007ffff637e981 in __GI_ppoll (fds=0x7fff24000ac8, nfds=1, timeout=<optimized out>, sigmask=0x0) at ../sysdeps/unix/sysv/linux/ppoll.c:39
#1  0x00005555562934c1 in ?? ()
#2  0x0000555556294a99 in ?? ()
#3  0x000055555624245f in ?? ()
#4  0x00005555560e2956 in ?? ()
#5  0x00005555560e4c25 in ?? ()
#6  0x00007ffff79bb4a4 in start_thread (arg=0x7fff467b1700) at pthread_create.c:456
#7  0x00007ffff6387d0f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97

Thread 20 (Thread 0x7fff47420700 (LWP 2308)):
#0  0x00007ffff637e981 in __GI_ppoll (fds=0x7fff20000ac8, nfds=1, timeout=<optimized out>, sigmask=0x0) at ../sysdeps/unix/sysv/linux/ppoll.c:39
#1  0x00005555562934c1 in ?? ()
#2  0x0000555556294a99 in ?? ()
#3  0x000055555624245f in ?? ()
#4  0x00005555560e2956 in ?? ()
#5  0x00005555560e4c25 in ?? ()
#6  0x00007ffff79bb4a4 in start_thread (arg=0x7fff47420700) at pthread_create.c:456
#7  0x00007ffff6387d0f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97

Thread 19 (Thread 0x7fff47c21700 (LWP 2307)):
#0  pthread_cond_timedwait@@GLIBC_2.3.2 () at ../sysdeps/unix/sysv/linux/x86_64/pthread_cond_timedwait.S:225
#1  0x000055555588049b in ?? ()
#2  0x000055555586f885 in ?? ()
#3  0x000055555588effb in ?? ()
#4  0x0000555556afccbf in ?? ()
#5  0x00007ffff79bb4a4 in start_thread (arg=0x7fff47c21700) at pthread_create.c:456
#6  0x00007ffff6387d0f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97

Thread 18 (Thread 0x7fff48422700 (LWP 2306)):
#0  0x00007ffff79c451f in __libc_recv (fd=42, buf=0x7fff48420ecc, n=4, flags=0) at ../sysdeps/unix/sysv/linux/x86_64/recv.c:28
#1  0x0000555555be958a in ?? ()
#2  0x0000555555beee50 in ?? ()
#3  0x0000555555bf0525 in ?? ()
#4  0x000055555588af61 in ?? ()
#5  0x000055555588b510 in ?? ()
#6  0x000055555588c285 in ?? ()
#7  0x00005555558948c7 in ?? ()
#8  0x000055555586f885 in ?? ()`

**Comment by MarcoFalke** [comment]
Hmm, looks like you are missing the debug symbols. You might want to try to download them into the same folder where your `bitcoin-qt` is located and then start `gdb` again (it should then print something like `Reading symbols from ./bin/bitcoin-qt...Reading symbols from ./bin/bitcoin-qt.dbg...done.`


```sh
wget https://github.com/achow101/bitcoin/releases/download/v0.18.0rc2/bitcoin-0.18.0rc2-x86_64-linux-gnu-debug.tar.gz
echo '438c107c985c88fa7e2eec13f7c03cdfedf079698ef7ea07597962fad080b954  bitcoin-0.18.0rc2-x86_64-linux-gnu-debug.tar.gz' | sha256sum --check
# untar and repeat above steps with gdb

**Comment by farrilis** [comment]
result:

```
Thread 1 "bitcoin-qt" received signal SIGINT, Interrupt.
0x00007ffff637e981 in __GI_ppoll (fds=0x555557f53618, nfds=2, timeout=<optimized out>, sigmask=0x0) at ../sysdeps/unix/sysv/linux/ppoll.c:39
39      ../sysdeps/unix/sysv/linux/ppoll.c: No such file or directory.
```

**Comment by promag** [comment]
> Hence maybe a prune=550 issue.

@farrilis but what happens without prune in 0.18?

**Comment by farrilis** [comment]
without prune I don't see this issue using 0.18. My full NETWORK_NODE is in a different VM.

**Comment by bitcoinhodler** [comment]
In 0.18.0rc3, the new `getrpcinfo` returns JSON with a key "duration", but there is nothing in that key name or in `bitcoin-cli help getrpcinfo` that tells me what time units it's counting in. Is that duration in seconds? Milliseconds? I have no idea.

**Comment by nitramiz** [comment]
tACK 0.18.0rc2 on MacOS 10.13.6:
* Built a docker image for bitcoind (blockstream/bitcoind@sha256:4c072542ff148b99467cab32756c3c2cdd1e05a8b8e30c2b1fe28a9ca90ec0b8) from the binary at bitcoincore.org (`x86_64-linux-gnu.tar.gz`)
* Synced testnet from scratch and reindexed once 
* Tested createfunded/decode/finalize PSBT, importmulti and broadcasting a tx (with the help of HWI and a Trezor)

**Comment by promag** [comment]
@bitcoinhodler it is microseconds. `getrpcinfo` help message was completed in #15754.

**Comment by jonatack** [comment]
ACK bitcoin-0.18.0rc3-x86_64-linux-gnu.tar.gz

Linux Debian 4.19.28-2 (2019-03-15) x86_64 GNU/Linux

    tested installing
    tested loading wallets
    tested bitcoind with nearly up-to-date mainnet blocks, synced rapidly without issue
    tested bitcoin-qt, verifying debug file, windows info/console/network/peers, and running cli commands in the console

**Comment by cfromknecht** [comment]
A user recently reported https://github.com/lightningnetwork/lnd/issues/2961 in testing lnd with an 0.18-ish build. I was able to locate https://github.com/bitcoin/bitcoin/pull/13541 which introduces a breaking change to the `sendrawtransaction` API, preventing lnd from publishing transactions with the following error:
> Second argument must be numeric (maxfeerate) and no longer supports a boolean. To allow a transaction with high fees, set maxfeerate to 0.

In skimming the draft release notes I could not find anything mentioning this change, is it worth adding?

The change affects all prior versions of lnd including 0.6 tagged today, so we'll be working on an update to our rpcclient to support the new numeric value and support downgrading to the boolean value on pre-0.18 nodes in our next 0.6.1 release.

(happy to make a separate issue if there's more to be done besides updating release notes)

**Comment by fanquake** [comment]
@cfromknecht #13541 isn't part of `v0.18`, (it's currently in `master` and will be released in `v0.19`), so I'd assume that user is running some version of `master` they've compiled themselves? Although looking at the related issue, they originally said they were running `v0.17.1`, so it's a bit hard to tell?

**Comment by cfromknecht** [comment]
@fanquake

> #13541 isn't part of v0.18, (it's currently in master and will be released in v0.19)

oh i see, that then makes perfect sense why i couldn't find it in the release notes :)
thank you for the clarification, that should give us plenty of time to support the change from our end

> they originally said they were running v0.17.1

i presume it's because the versions may not be bumped in their branch, but yes it is unclear

**Comment by molxyz** [comment]
@Sjors Hi, thank you for this thread. I'm also trying to test 0.18.0rc4. 

I have a question if anyone knows what the button `Main Window` on bitcoin-qt is for? I am running v0.18.0rc4, started seeing this button since last RC version, could not get it to do anything, thought it would be fixed in RC4 but still the same. I have tried by going from one window to another and clicked on this button `Main Window` but it doesn't do anything so I am wondering why it's there. 

I compiled the software from source on WSL using bitcoin guide `https://github.com/bitcoin/bitcoin/blob/master/doc/build-windows.md` and running it on Windows 10.

![image](https://user-images.githubusercontent.com/21127776/56603671-28948080-65ce-11e9-8dbf-2a1925fb69da.png)

**Comment by Sjors** [comment]
@molxyz please create a new issue for that; this issue is for finding potential release blocking problems. I think it was introduced [here](https://github.com/bitcoin/bitcoin/pull/14573#issuecomment-438704781).

**Comment by molxyz** [comment]
@Sjors  Thank you. Will do right now.

**Comment by fanquake** [comment]
Closing this now that `v0.18.0` has been released.
